vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -75,6 +88,64 @@ func FromUUID(client *govmomi.Client, uuid string) (*object.VirtualMachine, erro
 	return vm.(*object.VirtualMachine), nil
 }
 
+// virtualMachineFromSearchIndex gets the virtual machine reference via the
+// SearchIndex MO and is the method used to fetch UUIDs on newer versions of
+// vSphere.
+func virtualMachineFromSearchIndex(client *govmomi.Client, uuid string) (object.Reference, error) {
+	log.Printf("[DEBUG] Using SearchIndex to look up UUID %q", uuid)
+	search := object.NewSearchIndex(client.Client)
+	ctx, cancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
+	defer cancel()
+	result, err := search.FindByUuid(ctx, nil, uuid, true, structure.BoolPtr(false))
+	if err != nil {
+		return nil, err
+	}
+
+	if result == nil {
+		return nil, newUUIDNotFoundError(fmt.Sprintf("virtual machine with UUID %q not found", uuid))
+	}
+
+	return result, nil
+}
+
+// virtualMachineFromContainerView is a compatability method that is
+// used when the version of vSphere is too old to support using SearchIndex's
+// FindByUuid method correctly. This is mainly to facilitate the ability to use
+// FromUUID to find both templates in addition to virtual machines, which
+// historically was not supported by FindByUuid.
+func virtualMachineFromContainerView(client *govmomi.Client, uuid string) (object.Reference, error) {
+	log.Printf("[DEBUG] Using ContainerView to look up UUID %q", uuid)
+	m := view.NewManager(client.Client)
+
+	vctx, vcancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
+	defer vcancel()
+	v, err := m.CreateContainerView(vctx, client.ServiceContent.RootFolder, []string{"VirtualMachine"}, true)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		dctx, dcancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
+		defer dcancel()
+		v.Destroy(dctx)
```

Comment:
If you don't care about the error, explicitly discard it, ie: `_ = v.Destroy` so it doesn't look like an oversight.

Reply:
This will actually produce an error if done:

```
# github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/virtualmachine
vsphere/internal/helper/virtualmachine/virtual_machine_helper.go:130:5: no new variables on left side of :=
make: *** [GNUmakefile:9: build] Error 2
```